lib/rom/memory/dataset.rb

```diff
@@ -46,9 +46,21 @@ def project(*names)
 
       # Sort a dataset
       #
+      # @param [Array<Symbol>] names
+      #   Names of fields to order tuples by
+      #
+      # @option [Boolean] :nils_first (false)
+      #   Whether `nil` values should be placed before others
+      #
       # @api public
-      def order(*names)
-        sort_by { |tuple| tuple.values_at(*names) }
+      def order(*fields)
+        nils_first = fields.pop[:nils_first] if fields.last.is_a?(Hash)
+
+        sort do |a, b|
+          fields # finds the first difference between selected fields of tuples
+            .map { |n| __compare a[n], b[n], nils_first }
+            .detect { |r| r != 0 } || 0
```

Comment:
`<=>` (and `__compare`) returns `nil` when `a[n]` and `b[n]` are incomparable (against my expectation that they should be as values from the same field). In that case the whole ordering should report incompatibility. If I applied `nil.nonzero?` it would raise `NoMethodError` instead.

Maybe this edge case should be covered too. Mea culpa.

And wow! I tried `.detect(0) { ... }` and couldn't understand why it failed. I sooo like `.detect(->{0}) { ... }`!